src/mainwindow.cpp

```diff
@@ -1472,6 +1472,7 @@ void MainWindow::addNewEvent(QString event_type)
     if (editor && editor->project) {
         DraggablePixmapItem *object = editor->addNewEvent(event_type);
         if (object) {
+            updateObjects();
```

Comment:
Yeah this was the best solution I was able to come up with as well. I don't love it because it causes `updateSelectedObjects()` to get get called a bunch more times for every new object added, but this is a solution and I didn't notice any real life performance impact.